homeassistant/components/smarttub/sensor.py

```diff
@@ -0,0 +1,35 @@
+"""Platform for sensor integration."""
+import logging
+
+from .const import DOMAIN, SMARTTUB_CONTROLLER
+from .entity import SmartTubEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up climate entity for the thermostat in the tub."""
+
+    controller = hass.data[DOMAIN][entry.entry_id][SMARTTUB_CONTROLLER]
+
+    entities = [SmartTubState(controller.coordinator, spa) for spa in controller.spas]
+
+    async_add_entities(entities)
+
+
+class SmartTubState(SmartTubEntity):
+    """The state of the spa."""
+
+    def __init__(self, coordinator, spa):
+        """Initialize the entity."""
+        super().__init__(coordinator, spa, "state")
+
+    @property
```

Comment:
Can this be moved into the base class?

Reply:
It can. It won't be appropriate for all of the entities (e.g. there are multiple pumps and multiple lights) but I can deal with that in those subclasses.